main.py

```diff
@@ -1,121 +1,131 @@
-    w_desc = data["weather"][0]["description"]
-    pressure = main["pressure"]
-    visibility = data["visibility"]
-    visibility_new = visibility / 1000
-    wind = data["wind"]["speed"]
-
-    def printData():
-        breakliner()
-        print(f"{CITY2:-^30}")
-        print(f"Weather: {w_main}:- {w_desc}")
-        ktc(temp, temp_min, temp_max)
-        print(f"Humidity: {humidity}%")
-        print(f"Pressure: {pressure} hPa")
-        print(f"Wind speed: {wind} m/s")
-        print(f"Visibility: {visibility} m (or) {visibility_new} km")
-
-    printData()
-        
-def OWMCITY():
-    # City Name request
-    global CITY
-    CITY = input(
-        "Enter City Name (should be in compliance to OpenWeatherMap's City Index): "
-    )
-    if CITY == "exit":
-        clearScreen()
-        print("Three")
-        sleep(1)
-        clearScreen()
-        print("Two")
-        sleep(1)
-        clearScreen()
-        print("One.")
-        sleep(1)
-        clearScreen()
-        print("Exiting...")
-        sleep(0.5)
-        exit()
-
-    else:
-        pass
-    # The URL in actuality be like:
-    URL = BASE_URL + "q=" + CITY + "&appid=" + TOKEN
-
-    # PROD: Request Response
-    global resp_PROD
-    resp_PROD = requests.get(URL)
-    if resp_PROD.status_code == 200:
-        pass
-    else:
-        raise TypeError("Oops, wrong city name or code?")
-
-    global data
-    data = resp_PROD.json()
-    
-    data_processing()
-    exitQuery()
-
-OWMCITY()
+from __future__ import annotations
+
+from platform import system
+from tkinter import Menu, Tk, messagebox
+from tkinter.ttk import Button, Entry, Frame, Label, Style
+
+from requests import Response, get
+
+
+def self_return_decorator(func, *args, **kwargs):
+    # Allows for chaining of methods
+    # Example:
+    # App.OWMCITY().method1().method2()
+    # This does assume however that the methods don't have any return values
+    # If they do, then the return value will not be returned
+    # This is because the return value is the App object, not the return value of the method
+    def self_returner(self=None, *args, **kwargs):
+        func(self, *args, **kwargs)
+        return self
+
+    return self_returner
+
+
+class App(Tk):
+    def __init__(self):
+        super().__init__()
+        self.withdraw()
+
+        # Set up Menubar
+        if system() == "Darwin":
+            self.menubar = Menu(self)
+            # Apple menus have special names and special commands
+            self.app_menu = Menu(self.menubar, tearoff=0, name="apple")
+            self.menubar.add_cascade(label="App", menu=self.app_menu)
+        else:
+            self.menubar = Menu(self)
+            self.app_menu = Menu(self.menubar, tearoff=0)
+            self.menubar.add_cascade(label="App", menu=self.app_menu)
+        self.menubar.add_command(label="About Weather", command=self.about)
+        self.config(menu=self.menubar)
+
+        Style().theme_use("clam")
```

Comment:
Cause on MacOS, its *ugly*.

Reply:
I wouldn't say the Windows theme is pretty either, but it's the native and so it's not a really good idea to force a theme that's completely different from any other apps.
And the macOS theme also has dark mode :D

Reply:
Plus, imo, a uniform look across systems is nice.

Reply:
> Plus, imo, a uniform look across systems is nice.

I'd argue that :D
Imo it's better to have native look rather than uniform look across platforms.

Reply:
Well, I agree with both. Maybe we can add OS-native UI while keeping the UI similar across OS builds. Best of both worlds?

Reply:
Well the UI is going to be similar just not have a uniform look. If y'all want native looks then we can do that.